prymer/api/oligo.py

```diff
@@ -106,58 +106,77 @@ class Oligo(OligoLike, Metric["Oligo"]):
     likely to form stable hairpins or dimers, leading to reduced efficiency of the reaction.
 
     Attributes:
+        bases: the base sequence of the oligo (excluding any tail)
         tm: the calculated melting temperature of the oligo
-        penalty: the penalty or score for the oligo
         span: the mapping of the primer to the genome
+        penalty: the penalty or score for the oligo
         name: an optional name to use for the primer
         tm_homodimer: calculated melting temperature that represents
             the tendency of an oligo to bind to itself (self-complementarity)
         tm_3p_anchored_homodimer: calculated melting temperature that represents
             the tendency of a primer to bind to the 3'-END of an identical primer
         tm_secondary_structure: calculated melting temperature of the oligo hairpin structure
-        bases: the base sequence of the oligo (excluding any tail)
         tail: an optional tail sequence to put on the 5' end of the primer
 
     """
 
+    bases: str
     tm: float
-    penalty: float
     span: Span
+    penalty: float
+    name: Optional[str] = None
     tm_homodimer: Optional[float] = None
     tm_3p_anchored_homodimer: Optional[float] = None
     tm_secondary_structure: Optional[float] = None
-    bases: Optional[str] = None
     tail: Optional[str] = None
 
     def __post_init__(self) -> None:
-        super(Oligo, self).__post_init__()
+        if len(self.bases) == 0:
+            raise ValueError("Bases must not be an empty string.")
+        elif self.span.length != len(self.bases):
```

Comment:
Do we ever have a situation where the primer is targeting an alternate haplotype with an indel so then lengths would differ?

Reply:
It's theoretically possible .. but I'd suggest we relax this constraint the first time that comes up, as it's more likely a bug.